File: python/leap-year.py
def checkLeap(year):
    
    if(year<1):
        #Checks so that only a positive value is entered.
        print("That year is invalid")
    else:
        #Leap year Algorithm
        if(year % 4 == 0 and year % 100 != 0 or year % 400 == 0):
            print(year," is a leap year." ) # It is a leap year
        else:
            print(year," is not a leap year." )# not a leap year


year = int(input("Enter year to check if it is leap year: "))


checkLeap(year)            

File: javascript/leap-year.js
function checkLeap(year){
 this.year = year;
if(year<1){
    //Checks so that only a positive value is entered.
    console.log(year, "is not a valid year");
}
else{
 //Leap year Algorithm
if (year % 4 == 0 && year % 100 != 0 || year % 400 == 0) {
    return 1; // It is a leap year
    } else {
    return 0; // not a leap year
    }
}
}
console.log(checkLeap(2022));